### ISSUE TITLE
Detect when something cannot be a function type

### ISSUE THREAD

**Issue by GoogleCodeExporter** [agda/agda#437]
```agda
record Unit : Set where
data Bool : Set where true false : Bool

F : Bool -> Set
F true  = Bool
F false = Unit

f : (b : Bool) -> F b
f true  = true
f false = record {}

-- this should give an error, but only gives yellow
test : Bool
test = f _ false
```
Using the "constructor-headed" facility, Agda could detect that the type of f _ 
is never a function type and report an error here.  This is not done because 
"forcePi" is deactivated in Term.hs.  force Pi would synthesize a function type 
(x : X) -> Y x for the type of f _ and then try to solve for X and Y, failing. 

Reactivating forcePi would involve meta-variables for irrelevance, so it is not a trivial fix...

Original issue reported on code.google.com by `andreas....@gmail.com` on 22 Aug 2011 at 2:08

**Comment by nad** [comment]
@UlfNorell, I'm guessing that your recent code sprint has fixed this issue. I added a test case.

**Comment by andreasabel** [comment]
Possibly fixed by b1473f66a6b58b0784c894c202f5f2114f8dac29 .
